src/lib.rs

```diff
@@ -129,17 +129,17 @@ pub extern "C" fn bind(
 
     let mut sock = lock.write().unwrap();
 
-    match *sock {
-        Socket::Created => (),
-        _ => return error(libc::EBADFD, -1),
-    }
-
-    if next!(bind(fd, addr, len)) == -1 {
-        return -1;
+    if let Socket::Created = *sock {
+        match next!(bind(fd, addr, len)) {
+            0 => {
+                *sock = Socket::Bound(Parameters { client: false });
+                return 0;
+            },
+            r => r,
+        };
```

Comment:
Remove semi-colon.

Reply:
When I removed the semi-colon, `cargo build` would fail, how to avoid that?

Reply:
Oops. Keep the semi-colon. Put `return` immediately before `match`.

Reply:
If so, will it be good to change `r => r` to `r => error(libc::EBADFD, -1)`? This makes it even more simple and clear.

Reply:
No. `r` is the error return value from `bind()`. We want to propagate that upward. We only return `error(libc::EBADFD, -1)` when the socked is in a bad state.